lib/Ebook.php

```diff
@@ -48,71 +65,574 @@ class Ebook{
+	/** @var array<Contributor> $_Illustrators */
+	protected $_Illustrators = null;
+	/** @var array<Contributor> $_Translators */
+	protected $_Translators = null;
+	/** @var array<Contributor> $_Contributors */
+	protected $_Contributors = null;
+	/** @var ?array<string> $_TocEntries */
+	protected $_TocEntries = null; // A list of non-Roman ToC entries ONLY IF the work has the 'se:is-a-collection' metadata element, null otherwise
+	protected ?string $_Url = null;
+	protected ?bool $_HasDownloads = null;
+	protected ?string $_UrlSafeIdentifier = null;
+	protected ?string $_HeroImageUrl = null;
+	protected ?string $_HeroImageAvifUrl = null;
+	protected ?string $_HeroImage2xUrl = null;
+	protected ?string $_HeroImage2xAvifUrl = null;
+	protected ?string $_CoverImageUrl = null;
+	protected ?string $_CoverImageAvifUrl = null;
+	protected ?string $_CoverImage2xUrl = null;
+	protected ?string $_CoverImage2xAvifUrl = null;
+	protected ?string $_ReadingEaseDescription = null;
+	protected ?string $_ReadingTime = null;
+	protected ?string $_AuthorsHtml = null;
+	protected ?string $_AuthorsUrl = null; // This is a single URL even if there are multiple authors; for example, /ebooks/karl-marx_friedrich-engels/
+	protected ?string $_ContributorsHtml = null;
+	protected ?string $_TitleWithCreditsHtml = null;
+	protected ?string $_TextUrl = null;
+	protected ?string $_TextSinglePageUrl = null;
+	protected ?string $_TextSinglePageSizeFormatted = null;
+	protected ?string $_IndexableText = null;
+
+	// *******
+	// GETTERS
+	// *******
+
+	/**
+	 * @return array<GitCommit>
+	 */
+	protected function GetGitCommits(): array{
+		if($this->_GitCommits === null){
+			$this->_GitCommits = Db::Query('
+							SELECT *
+							from GitCommits
+							where EbookId = ?
+							order by Created desc
+						', [$this->EbookId], GitCommit::class);
+		}
+
+		return $this->_GitCommits;
+	}
+
+	/**
+	 * @return array<EbookTag>
+	 */
+	protected function GetTags(): array{
+		if($this->_Tags === null){
+			$this->_Tags = Db::Query('
+						SELECT t.*
+						from Tags t
+						inner join EbookTags et using (TagId)
+						where EbookId = ?
+						order by et.EbookTagId
+					', [$this->EbookId], EbookTag::class);
+		}
+
+		return $this->_Tags;
+	}
+
+	/**
+	 * @return array<LocSubject>
+	 */
+	protected function GetLocSubjects(): array{
+		if($this->_LocSubjects === null){
+			$this->_LocSubjects = Db::Query('
+							SELECT l.*
+							from LocSubjects l
+							inner join EbookLocSubjects el using (LocSubjectId)
+							where EbookId = ?
+							order by el.EbookLocSubjectId
+					', [$this->EbookId], LocSubject::class);
+		}
+
+		return $this->_LocSubjects;
+	}
+
+	/**
+	 * @return array<CollectionMembership>
+	 */
+	protected function GetCollectionMemberships(): array{
+		if($this->_CollectionMemberships === null){
+			$this->_CollectionMemberships = Db::Query('
+							SELECT *
+							from CollectionEbooks
+							where EbookId = ?
+							order by CollectionEbookId
```

Comment:
As you pointed out below, I removed `CollectionEbookId` and added a `SortOrder` to this table because that's what we want to sort on, not `SequenceNumber`. The `SortOrder` presents the collection memberships in the same order the producer added them to `content.opf`, e.g. for *To the Lighthouse*:

№ 15 in the Modern Library’s 100 Best Novels set.  
Part of the Encyclopædia Britannica’s Great Books of the Western World set.  
№ 2 in the BBC’s 100 Greatest British Novels (2015) set.

(The sequence numbers there are 15, `NULL`, 2, so they're not what we want to sort on here. We do sort by `SequenceNumber` in `collections/get.php` to present the ebooks in the right order in a collection.)